Kernel/DoubleBuffer.cpp

```diff
@@ -77,4 +77,24 @@ ssize_t DoubleBuffer::read(UserOrKernelBuffer& data, size_t size)
     return (ssize_t)nread;
 }
 
+ssize_t DoubleBuffer::peek(UserOrKernelBuffer& data, size_t size)
+{
+    if (!size || m_storage.is_null())
+        return 0;
+    VERIFY(size > 0);
+    Locker locker(m_lock);
+    if (m_read_buffer_index >= m_read_buffer->size && m_write_buffer->size != 0) {
+        flip();
+    }
+    if (m_read_buffer_index >= m_read_buffer->size)
+        return 0;
+    size_t nread = min(m_read_buffer->size - m_read_buffer_index, size);
+    if (!data.write(m_read_buffer->data + m_read_buffer_index, nread))
+        return -EFAULT;
+    compute_lockfree_metadata();
+    if (m_unblock_callback && m_space_for_writing > 0)
+        m_unblock_callback();
+    return (ssize_t)nread;
 }
+
+}
```

Comment:
This won't pass lint, please make sure to add newlines on the last line of files.